source/opaque.d.ts

```diff
@@ -151,18 +156,59 @@ function getMoneyForAccount(accountNumber: AccountNumber): AccountBalance {
 getMoneyForAccount(createAccountNumber());
 
 // But this won't, because it has to be explicitly passed as an `AccountNumber` type!
+// Critically, you could not accidentally use an `AccountBalance` as an `AccountNumber`.
 getMoneyForAccount(2);
 
-// You can use opaque values like they aren't opaque too.
-const accountNumber = createAccountNumber();
+// You can also use tagged values like their underlying, untagged type.
+// I.e., this will compile successfully because an `AccountNumber` can be used as a regular `number`.
+// In this sense, the underlying base type is not hidden, which differentiates tagged types from opaque types in other languages.
+const accountNumber = createAccountNumber() + 2;
+```
 
-// This will compile successfully.
-const newAccountNumber = accountNumber + 2;
+@example
+```
+import type {Tagged} from 'type-fest';
+
+// You can apply multiple tags to a type by using `Tagged` repeatedly.
+type Url = Tagged<string, 'URL'>;
+type SpecialCacheKey = Tagged<Url, 'SpecialCacheKey'>;
+
+// You can also pass a union of tag names, so this is equivalent to the above, although it doesn't give you the ability to assign distinct metadata to each tag.
+type SpecialCacheKey2 = Tagged<string, 'URL' | 'SpecialCacheKey'>;
+```
+
+@category Type
+*/
+export type Tagged<Type, Tag extends PropertyKey, TagMetadata = never> = Type & MultiTagContainer<Tag, TagMetadata>;
```

Comment:
I ended up making a tag's default metadata `never` instead of `void`, as `never` has more consistent/sensible assignability semantics